src/spider/worker/worker.cpp

```diff
@@ -188,6 +188,42 @@ auto get_args_buffers(spider::core::Task const& task)
     return args_buffers;
 }
 
+auto get_args_buffers(
```

Comment:
1. Overloading this function is confusing as they both take a `Task`.
2. Lets try to improve the name since it isn't very obvious how `get_args_buffers` is "getting task details" (as stated in the PR title).
    * on a side note, IMO `get_arg_buffers` (get argument buffers) reads a bit better than `args`
        * for `args` "get arguments' buffers" makes sense, but "get arguments buffers" doesn't so it can feel a bit weird